runtime/shared_common/ClassDebugDataProvider.cpp

```diff
@@ -686,6 +687,21 @@ ClassDebugDataProvider::setPermission(J9VMThread* currentThread, AbstractMemoryP
 	if (pageSize > 0) {
 		bool doVerbosePages = permSetter->isVerbosePages();
 		UDATA pageProtectFlag = J9PORT_PAGE_PROTECT_READ;
+		
+		
+		if ((UnitTest::CLASS_DEBUG_DATA_TEST != UnitTest::unitTest)
+			&& (ROUND_DOWN_TO(pageSize, (UDATA)lntProtectLow) < (UDATA)GET_DEBUG_MEM_START(_theca))
+		) {
+			/* Do not change the memory protection of header page. The header page is handled by (un)protectHeaderReadWriteArea(). */
```

Comment:
I guess the reason for putting header in its own page is because of memory protection, which is done in pages. IIRC, in the past 5+ years, memory protection never helped us capture an overwrite to the SCC in PMRs. I start to think whether we should still keep memory protection on by default.

Reply:
Any overwrite likely comes from user code, and these issues wouldn't come to your attention, and may not even be reported to service as the user could diagnose them. I don't think we should disable memory protection. Making the header 0x1000 isn't a requirement for this feature and could be done later separately.